backend/functions/src/matching/functions.ts

```diff
@@ -91,7 +90,9 @@ async function makeMatches(courseId: string) {
   for (let i = 0; i < groupTriples.length; i += 3) {
     groupCounter += 1
     const newGroup = groupTriples.slice(i, i + 3)
+
     newGroups.push({
+      groupId: courseId + groupCounter + lastGroupNumber,
```

Comment:
Why do you add the `lastGroupNumber` here? The `courseId` and `groupCounter` should be enough to guarantee uniqueness.

Reply:
Should `groupCounter` even be used? That's a local variable, wouldn't it be reset every time they make new matches, with the `courseId` staying the same?

Reply:
oh I didn't realize that was the local variable. there should be some what of indicating which group it is lol

Reply:
Actually I see how `groupCounter` and `lastGroupNumber` work together to represent this group's current number, since `lastGroupNumber` isn't incremented when new groups are added. Could be clearer if these were just combined into one thing